src/app/pages/wallets/my-wallets/my-wallet/pages-wallets-my-wallets-my-wallet.component.ts

```diff
@@ -8,4 +10,24 @@ import { MyWallet } from '../pages-wallets-my-wallet.model';
 })
 export class PagesWalletsMyWalletsMyWalletComponent {
   @Input() public wallet!: MyWallet;
-}
+
+  constructor(
+    public dialog: MatDialog,
+  ) {}
+
+  openEditWalletDialog() {
```

Comment:
You should make this component dumb / presentational and move this logic to parent component. In this component you should just emit event like `walletEdit` using of course `EventEmitter`. That way you will have all the logic inside parent component and only parent component will be responsible for opening wallet modal.